Makefile

```diff
@@ -19,74 +45,126 @@ all: help
+
+.PHONY: debug
+debug:
+	@-echo "Following variables are used:"
+	@-echo "  SHELL:                         $(SHELL)"
+	@-echo "  CHECK:                         $(CHECK)"
+	@-echo "  APP_ROOT:                      $(APP_ROOT)"
+	@-echo "  APP_NAME:                      $(APP_NAME)"
+	@-echo "  BUMP_XARGS:                    $(BUMP_XARGS)"
+	@-echo "  CONDA_HOME:                    $(CONDA_HOME)"
+	@-echo "  CONDA_BIN:                     $(CONDA_BIN)"
+	@-echo "  CONDA_ENV:                     $(CONDA_ENV)"
+	@-echo "  CONDA_MSG:                     $(CONDA_MSG)"
+	@-echo "  CONDA_CMD:                     $(CONDA_CMD)"
+	@-echo "  CONDA_TARGET_PREFIX [literal]: $(CONDA_TARGET_PREFIX)"
+	@-echo "  CONDA_ACTUAL_PREFIX [called]:  $(shell $(CONDA_TARGET_PREFIX))"
+	@-echo "  CONDA_TARGET_PREFIX [literal]: $(CONDA_ACTUAL_PREFIX)"
+	@-echo "  CONDA_ACTUAL_PREFIX [called]:  $(shell $(CONDA_ACTUAL_PREFIX))"
+	@-echo "  DOCKER_TAG:                    $(DOCKER_TAG)"
 
 ## Conda targets
 
-.PHONY: check_conda
-check_conda:
-ifndef CONDA
-		$(error "Conda is not available. Please install miniconda: https://conda.io/miniconda.html")
+.PHONY: conda-check
+conda-check:
+ifndef CONDA_BIN
+	$(error "Conda is not available. Please install miniconda: https://conda.io/miniconda.html")
 endif
 
-.PHONY: conda_env
-conda_env: check_conda
+.PHONY: conda-update
+conda-update: conda-check
 	@echo "Updating conda environment $(CONDA_ENV) ..."
-	"$(CONDA)" env update -n $(CONDA_ENV) -f environment.yml
+	@-"$(CONDA_BIN)" env update -n "$(CONDA_ENV)" -f environment.yml
 
-.PHONY: envclean
-envclean: check_conda
+.PHONY: conda-clean
+conda-clean: conda-check
 	@echo "Removing conda env $(CONDA_ENV)"
-	@-"$(CONDA)" remove -n $(CONDA_ENV) --yes --all
+	@-"$(CONDA_BIN)" remove -n "$(CONDA_ENV)" --yes --all
 
-.PHONY: spec
-spec: check_conda
+.PHONY: conda-spec
+spec: conda-check
 	@echo "Updating conda environment specification file ..."
-	@-"$(CONDA)" list -n $(CONDA_ENV) --explicit > spec-file.txt
+	@-"$(CONDA_BIN)" list -n "$(CONDA_ENV)" --explicit > spec-file.txt
+
+## Version targets
+
+.PHONY: bump_dep
+bump_dep:
+	@-bash -c '$(CONDA_CMD) pip install bump2version'
+
+.PHONY: bump
+bump: bump_dep
+	@-echo "Updating package version ..."
+	@[ "${VERSION}" ] || ( echo ">> 'VERSION' is not set"; exit 1 )
+	@-bash -c '$(CONDA_CMD) bump2version $(BUMP_XARGS) --new-version "${VERSION}" patch;'
 
 ## Build targets
 
 .PHONY: bootstrap
-bootstrap: check_conda conda_env bootstrap_dev
+bootstrap: conda-check conda-update install
 	@echo "Bootstrap ..."
 
-.PHONY: bootstrap_dev
-bootstrap_dev:
-	@echo "Installing development requirements for tests and docs ..."
-	@-bash -c "source $(ANACONDA_HOME)/bin/activate $(CONDA_ENV) && pip install -r requirements_dev.txt"
+.PHONY: bootstrap-dev
+bootstrap-dev: conda-check conda-update install-dev
+	@echo "Bootstrap for development ..."
 
 .PHONY: install
-install: bootstrap
+install:
```

Comment:
before `make install` was meant to install the whole app ... so would call `bootstrap` to create conda env. No one has to call `make bootstrap install`.

The usual Python installation can still be done manually ... the intention of the Makefile is to forget the details and avoid reading the docs ;)

https://twitcher.readthedocs.io/en/latest/installation.html

Reply:
Ok. We can revert this change to keep the original "full install" functionality.
I think it would be still good to add another target, (`quick-install`?) that does what was intended here.
The case I often encounter is that I update some code, run `make install`, and then have to wait forever for `conda env update` to do its thing, although there is nothing to update specifically in the env.

Reply:
I would prefer to keep the previous `install` style and add a new target the "quick-install". 

Usually you don't need to run `install` again ... just once. It uses `python setup.py develop`.

Reply:
Indeed, you are right about the need to install only once.
I have found here: https://stackoverflow.com/questions/19048732/python-setup-py-develop-vs-install
It is suggested to use `pip install -e <dir>` instead of `python setup.py develop` to allow properly installing packages requirements.
So maybe only need to use this command, and drop the "quick-install" idea altogether.

Reply:
Ok. Do you like to change it?

Reply:
Will do :)